toupee/data.py

```diff
@@ -205,6 +208,29 @@ def resample(self, sample_size:int=None, weights=None, replace:bool=True):
         self.resample_replace = replace
         self.__class__ = ResamplingDataset
         return self
+    
+    def copy_with_new_data(self, new_data):
+        """ Create a new copy but with new data. Useful for distillation and post-processing. """
+        new_dataset = copy.copy(self)
+        new_dataset.raw_data
+        new_dataset._process(**self.kwargs)
+        return new_dataset
+
+    def map(self, f, columns=None):
```

Comment:
you can instead use Pandas DataFrame.apply()

Reply:
🐼 🔪

Reply:
yeah I don't think it's a great move to add a requirement for pandas and then have to support pandas inputs just for that.

Reply:
Death to the panda(s)